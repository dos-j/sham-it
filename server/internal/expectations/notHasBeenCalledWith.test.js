const notHasBeenCalledWith = require("./notHasBeenCalledWith");

describe("unit: notHasBeenCalledWith", () => {
  let handler;
  let requestStore;

  beforeEach(() => {
    requestStore = [];
    handler = notHasBeenCalledWith(requestStore);
  });

  test("it should return 204 No Content if the query given matches the number of requests in the requestStore", () => {
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test1",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test2",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test3",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });

    expect(handler({ op: "==", prop: "pathname", value: "/test4" })).toEqual({
      status: 204
    });
  });

  test("it should return 417 Expectation failed if query given does not match the number of requests in the requestStore", () => {
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test1",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test2",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test3",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });

    expect(handler({ op: "==", prop: "pathname", value: "/test2" })).toEqual(
      expect.objectContaining({
        status: 417
      })
    );
  });

  test("the error message should match the snapshot", () => {
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test1",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test2",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test3",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });

    expect(
      handler({ op: "==", prop: "pathname", value: "/test2" })
    ).toMatchSnapshot();
  });
});
