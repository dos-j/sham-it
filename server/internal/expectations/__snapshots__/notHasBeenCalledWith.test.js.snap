// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit: notHasBeenCalledWith the error message should match the snapshot 1`] = `
Object {
  "body": "Expected sham not to have been called with:
  {
    \\"op\\": \\"==\\",
    \\"prop\\": \\"pathname\\",
    \\"value\\": \\"/test2\\"
  }
But it was called with:
  GET /test2?a=1234
    accept: application/json",
  "status": 417,
}
`;
